### ISSUE TITLE
can't build the container

### ISSUE THREAD

**Issue by orangebutblue** [freqtrade/frequi#289]
I've tried it both the with the current github repository and the 0.0.4 release, same result every time:

```
root@pi4:/opt/ui# docker-compose build
Building frequi
Step 1/14 : FROM node:15.9.0-alpine as ui-builder
 ---> e4b41e164238
Step 2/14 : RUN mkdir /app
 ---> Using cache
 ---> 10d23fb0ee5a
Step 3/14 : WORKDIR /app
 ---> Using cache
 ---> f0f5afe08e60
Step 4/14 : COPY package.json /app/package.json
 ---> Using cache
 ---> a7a6dd3f88f2
Step 5/14 : COPY yarn.lock /app/yarn.lock
 ---> Using cache
 ---> f83bcec2f5dc
Step 6/14 : RUN yarn
 ---> Running in 95f7352254b7
yarn install v1.22.5
[1/4] Resolving packages...
[2/4] Fetching packages...
info fsevents@2.3.2: The platform "linux" is incompatible with this module.
info "fsevents@2.3.2" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@1.2.13: The platform "linux" is incompatible with this module.
info "fsevents@1.2.13" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning " > bootstrap@4.6.0" has unmet peer dependency "jquery@1.9.1 - 3".
warning "vue-grid-layout > @interactjs/actions@1.10.3" has unmet peer dependency "@interactjs/core@1.10.3".
warning "vue-grid-layout > @interactjs/actions@1.10.3" has unmet peer dependency "@interactjs/utils@1.10.3".
warning "vue-grid-layout > @interactjs/auto-start@1.10.3" has unmet peer dependency "@interactjs/core@1.10.3".
warning "vue-grid-layout > @interactjs/auto-start@1.10.3" has unmet peer dependency "@interactjs/utils@1.10.3".
warning "vue-grid-layout > @interactjs/modifiers@1.10.3" has unmet peer dependency "@interactjs/core@1.10.3".
warning "vue-grid-layout > @interactjs/modifiers@1.10.3" has unmet peer dependency "@interactjs/utils@1.10.3".
warning "vue-grid-layout > @interactjs/modifiers > @interactjs/snappers@1.10.3" has unmet peer dependency "@interactjs/utils@1.10.3".
warning " > @typescript-eslint/eslint-plugin@2.34.0" has incorrect peer dependency "@typescript-eslint/parser@^2.0.0".
warning " > @vue/eslint-config-airbnb@5.3.0" has unmet peer dependency "eslint-plugin-import@^2.18.2".
warning "@vue/eslint-config-airbnb > eslint-import-resolver-webpack@0.13.0" has unmet peer dependency "webpack@>=1.11.0".
warning " > @vue/eslint-config-typescript@5.1.0" has incorrect peer dependency "@typescript-eslint/parser@^2.7.0".
warning " > eslint-config-airbnb@18.2.1" has unmet peer dependency "eslint-plugin-import@^2.22.1".
warning " > eslint-config-airbnb@18.2.1" has unmet peer dependency "eslint-plugin-jsx-a11y@^6.4.1".
warning " > eslint-config-airbnb@18.2.1" has unmet peer dependency "eslint-plugin-react@^7.21.5".
warning " > eslint-config-airbnb@18.2.1" has unmet peer dependency "eslint-plugin-react-hooks@^4 || ^3 || ^2.3.0 || ^1.7.0".
warning " > sass-loader@10.1.1" has unmet peer dependency "webpack@^4.36.0 || ^5.0.0".
[4/4] Building fresh packages...
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
error /app/node_modules/deasync: Command failed.
Exit code: 1
Command: node ./build.js
Arguments:
Directory: /app/node_modules/deasync
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@7.1.2
gyp info using node@15.9.0 | linux | arm
gyp ERR! find Python
gyp ERR! find Python Python is not set from command line or npm configuration
gyp ERR! find Python Python is not set from environment variable PYTHON
gyp ERR! find Python checking if "python3" can be used
gyp ERR! find Python - "python3" is not in PATH or produced an error
gyp ERR! find Python checking if "python" can be used
gyp ERR! find Python - "python" is not in PATH or produced an error
gyp ERR! find Python checking if "python2" can be used
gyp ERR! find Python - "python2" is not in PATH or produced an error
gyp ERR! find Python
gyp ERR! find Python **********************************************************
gyp ERR! find Python You need to install the latest version of Python.
gyp ERR! find Python Node-gyp should be able to find and use Python. If not,
gyp ERR! find Python you can try one of the following options:
gyp ERR! find Python - Use the switch --python="/path/to/pythonexecutable"
gyp ERR! find Python   (accepted by both node-gyp and npm)
gyp ERR! find Python - Set the environment variable PYTHON
gyp ERR! find Python - Set the npm configuration variable python:
gyp ERR! find Python   npm config set python "/path/to/pythonexecutable"
gyp ERR! find Python For more information consult the documentation at:
gyp ERR! find Python https://github.com/nodejs/node-gyp#installation
gyp ERR! find Python **********************************************************
gyp ERR! find Python
gyp ERR! configure error
gyp ERR! stack Error: Could not find any Python installation to use
gyp ERR! stack     at PythonFinder.fail (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:302:47)
gyp ERR! stack     at PythonFinder.runChecks (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:136:21)
gyp ERR! stack     at PythonFinder.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:179:16)
gyp ERR! stack     at PythonFinder.execFileCallback (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:266:16)
gyp ERR! stack     at exithandler (node:child_process:333:5)
gyp ERR! stack     at ChildProcess.errorhandler (node:child_process:345:5)
gyp ERR! stack     at ChildProcess.emit (node:events:378:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:288:12)
gyp ERR! stack     at onErrorNT (node:internal/child_process:480:16)
gyp ERR! stack     at processTicksAndRejections (node:internal/process/task_queues:81:21)
gyp ERR! System Linux 5.10.4-v7l+
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /app/node_modules/deasync
gyp ERR! node -v v15.9.0
gyp ERR! node-gyp -v v7.1.2
gyp ERR! not ok
Build failed
ERROR: Service 'frequi' failed to build: The command '/bin/sh -c yarn' returned a non-zero code: 1

```
On a RPi4 if that wasn't clear from the hostname :)

**Comment by xmatthias** [comment]
i know there's problems building on a raspberry, but i don't know why.
the most likely culprit looks to be node-gpy - so maybe have a look at their documentation on how to properly build this ?

**Comment by orangebutblue** [comment]
I didn't realized that frequi is already integrated in freqtrade. I could successfully build that one instead and it works fine

**Comment by xmatthias** [comment]
yeah, so i guess you'd only need this for development builds ... but you won't develop on a raspberry most likely - so for now it's an issue i'll be happy to live with :laughing: 

hope that works for you, too

**Comment by uyplayer** [comment]
i  have the same issue , how to solve this ?

**Comment by orangebutblue** [comment]
Just  use freqtrade. the frequi is already integrated in the container

**Comment by xmatthias** [comment]
Building the container on a raspberry does not work - and there is currently no solution to this.

**Comment by xmatthias** [comment]
Seems like deasync requires python (and C make tools) to build in certain circumestances (depending on the node version, i suspect).

closed in cd91841
